### DISCUSSION TITLE
Performing Updates using the Docker Hub Image

### DISCUSSION THREAD

**Discussion by mengelsen** [cupcakearmy/cryptgeon#131] [Q&A] [answered]
If I modified the `docker-compose.yaml` file (to set [Environment Variables](https://github.com/cupcakearmy/cryptgeon?tab=readme-ov-file#environment-variables)), do I need to be concerned about loosing these changes when I `pull` the latest version of Cryptgeon from [Docker Hub](https://hub.docker.com/r/cupcakearmy/cryptgeon)?

Also, if I want to update Cryptgeon to the latest build, is this the process that I need to follow:

1.  List current images: `docker images`
2. Stop the current containers: `docker stop <container_id1> <container_id2> <container_id3>`
3. Remove the container: `docker rm <container_id1> <container_id2> <container_id3>)`
4. Pull the latest build of Cryptgeon: `docker pull cupcakearmy/krypton`
5. CD to the directory that contains the docker-compose.yaml file: `cd /opt/cryptgeon/`
6. Start the container: `docker compose up -d`
7. Delete the old image file (from step 1): `docker rmi <image_id>`

**Comment by cupcakearmy** [comment, marked as answer]
If you are using docker-compose

```bash
docker compose pull
docker compose up -d
```

Or automatically using something like [watchtower](https://github.com/containrrr/watchtower)
